packages/handlersjs-http/lib/handlers/json-store-handler.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { MemoryStore } from '@digita-ai/handlersjs-core';
+import { HttpHandlerContext } from '../models/http-handler-context';
+import { HttpHandlerResponse } from '../models/http-handler-response';
+import { JsonStoreHandler } from './json-store-handler';
+
+describe('JsonStoreHandler', () => {
+
+  interface StoreInterface {
+    data: string[];
+  }
+
+  const inputData = [ 'abc', 'defghij', '123' ];
+  const inputData2 = [ ... inputData, '456' ];
+
+  let jsonStoreHandler: JsonStoreHandler<StoreInterface>;
+  let memoryStore: MemoryStore<StoreInterface>;
+  let requestContext: HttpHandlerContext;
+
+  beforeEach(() => {
+
+    memoryStore = new MemoryStore([ [ 'data', inputData ] ]);
+
+    jsonStoreHandler = new JsonStoreHandler('data', memoryStore);
+
+    requestContext = {
+      request: {
+        url: new URL('http://localhost/'),
+        method: 'GET',
+        headers: {},
+      },
+    };
+
+  });
+
+  describe('handle()', () => {
+
+    it('can GET the data correctly', async () => {
+
+      const response: HttpHandlerResponse = await jsonStoreHandler.handle(requestContext).toPromise();
+      const resultData: string[] = JSON.parse(response.body);
+
+      expect(resultData).toEqual(inputData);
+      expect(response.status).toEqual(200);
+
+    });
+
+    it('can not use a wrong HTTP method', async () => {
+
+      requestContext.request.method = 'POST';
+      const response: HttpHandlerResponse = await jsonStoreHandler.handle(requestContext).toPromise();
+
+      expect(response.status).toEqual(405);
+
+    });
+
+    it('returns "not found" if the data is not available', async () => {
+
+      await memoryStore.delete('data');
+      const response: HttpHandlerResponse = await jsonStoreHandler.handle(requestContext).toPromise();
+
+      expect(response.status).toEqual(404);
+
+    });
+
+    it('can GET updated data', async () => {
+
+      let response: HttpHandlerResponse = await jsonStoreHandler.handle(requestContext).toPromise();
+      let resultData: string[] = JSON.parse(response.body);
```

Comment:
If initiated immediately, variables should be consts. Instead of overwriting them, use new variables later on.